addons/sale_pdf_quote_builder/tests/test_pdf_quote_builder.py

```diff
@@ -57,6 +57,18 @@ def setUpClass(cls):
             'res_id': cls.product.id,
         })
 
+    def _create_so_form(self, **values):
+        """Default values limited to preexisting ones. No Command"""
+        so_form = Form(self.env['sale.order'])
+        default_values = {
+            'partner_id': self.partner,
+            **values
+        }
+        for field_name, value in default_values.items():
+            so_form[field_name] = value
+        so_form.save()
+        return so_form
```

Comment:
```suggestion
        return so_form.save()
```
that way you have the record directly :eyes:

Not sure this helper is very useful though :thinking:

Reply:
Do you mean, the `_create_so_form` helper is not very useful, or just returning the so form from the `save` method directly? 

For the former case, since the so template documents are added via onchange events, I need to simulate the creation of a form, and I need that in a few places to test that the onchange trigger code does not mess up the so.

For the latter case, I guess it's one line less, so I don't see why not we shouldn't do it. To me its useful comment 🤷‍♂️

Reply:
oh, no I though it was returning the "so form", not the record. I need the so form if a test later has to alter the so

Reply:
About my suggestion to remove the record instead of the form, it's ok like this since you do use the form (probably read too quickly when reviewing).
For the second comment, I'm indeed not sure the helper is very useful for the 3 little uses, but it's ok if you believe it's clearer like this.

Other question: do you need to save the form before returning it, I find it a bit strange to save a form twice in one of your tests :thinking:

Reply:
I find that It makes the test code clearer. But I could add a parameter `save` to `_create_so_form` (default to True). 

I know its redundant to do the first check/assert in the setup phase of the last test (`test_quotation_document_is_removed_if_unavailable`) but, IMO, its easier to understand what the setup is doing and for that I need the form to be saved to access the internal record.

What do you think, should I add a `save` parameter and remove the redundant check, or keep it like that knowing my point of view?

Reply:
Certainly not worth a `save` parameter, the util should stay simple.
Keep it like this then, it make sense if you need to save before accessing the record :ok_hand: